### PR TITLE
Add a pretty print option to the metrics web api

### LINKED ISSUE #4645: Add a JSON preatty print option to the web api
It would be nice to have an option like:

* pretty: When appending ?pretty to any request made, the JSON returned will be pretty formatted.

### PR DESCRIPTION
**Pull request by purbon**
Fixes #4645

### CODE REVIEW

**Comment by ph** [comment]
Can we have a small test for the `Logstash::Json` class?

**Comment by purbon** [comment]
@ph not sure how to differentiate between preatty and not pretty, any criteria you might think of?

**Comment by ph** [comment]
@purbon The json format with the pretty option should have line data formatting in it ("\n" ...)
I think is important to test it since we wrap the library in a common accessor, if we ever change jrjackson for something else we should make sure it support a pretty output.

**Comment by purbon** [comment]
ok, will add a test looking for CR characters. :smile:

**Comment by purbon** [comment]
Test added. :+1:

File: logstash-core/spec/logstash/json_spec.rb [hunk 1]
```diff
@@ -56,6 +56,20 @@
         expect(LogStash::Json.dump(array)).to eql(json_array)
       end
 
+      context "pretty print" do
+
+        let(:hash) { { "foo" => "bar", :zoo => 2 } }
+
+        it "should serialize with pretty print" do
+          pprint_json = LogStash::Json.dump(hash, :pretty => true)
+          expect(pprint_json).to include("\n")
```
Comment (ph) [inline logstash-core/spec/logstash/json_spec.rb]: I think I would make it stricter with comparing the results with the actual generated string, But I guess this work too?
```
irb(main):001:0> JrJackson::Base.generate({ "foo" => "bar", :zoo => 2 }, :pretty => true)
=> "{\n  \"foo\" : \"bar\",\n  \"zoo\" : 2\n}"
irb(main):002:0> JrJackson::Base.generate({ "foo" => "bar", :zoo => 2 }, :pretty => false)
=> "{\"foo\":\"bar\",\"zoo\":2}"

**Comment by ph** [comment]
Small, comment for the tests but this lgtm.

File: logstash-core/spec/logstash/json_spec.rb (hunk 1 above)
Comment (purbon) [inline logstash-core/spec/logstash/json_spec.rb]: I'm not sure if doing it strictly could actually leak to some specific implementation details, should all pretty print be the same? or just include CR? I mean for example in the indentation. For now I think I will leave it like this and we can always improve it later on.

**Pull request closed without merging**